xapian-core/cluster/cluster.cc

```diff
@@ -0,0 +1,391 @@
+/** @file cluster.cc
+ *  @brief Cluster API
+ */
+/* Copyright (C) 2010 Richard Boulton
+ * Copyright (C) 2016 Richhiey Thomas
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <config.h>
+
+#include "xapian/cluster.h"
+
+#include <xapian/error.h>
+#include <api/termlist.h>
+#include <debuglog.h>
+#include <omassert.h>
+
+#include <cmath>
+#include <unordered_map>
+#include <vector>
+
+using namespace Xapian;
+using namespace std;
+
+FreqSource::~FreqSource()
+{
+    LOGCALL_DTOR(API, "FreqSource");
+}
+
+Similarity::~Similarity()
+{
+    LOGCALL_DTOR(API, "Similarity");
+}
+
+Clusterer::~Clusterer()
+{
+    LOGCALL_DTOR(API, "Clusterer");
+}
+
+TermListGroup::TermListGroup(const MSet &docs) {
+    LOGCALL_CTOR(API, "TermListGroup()", docs);
```

Comment:
Yes - the logging code generates the parameter list, so currently the log will contain stuff like `TermListGroup()(MSet(...))`).